### PR TITLE
Enable item-backed add-ons

### PR DESCRIPTION
**Pull request by joannasese**
To create item-backed add-on:
```c#
var addon = plan.NewAddOn("item_code");
addon.UnitAmountInCents.Add("USD", 200);
addon.Create();
```

To get an item-backed-add-on:
```c#
plan.GetAddOn(item.ItemCode);
```

To update item-backed add-on:
```c#
addon.UnitAmountInCents["USD"] = 200;
addon.Update();
```

When add-on has an associated item, only the following attributes are updatable:
```c#
unit_amount_in_cents
default_quantity
display_quantity_on_hosted_page
optional
```

### CODE REVIEW

File: Library/AddOn.cs [hunk 1]
```diff
@@ -176,13 +204,27 @@ internal override void ReadXml(XmlTextReader reader)
             }
         }
 
-        internal override void WriteXml(XmlTextWriter xmlWriter)
+        internal void WriteItemBackedAddOnXml(XmlTextWriter xmlWriter)
         {
             xmlWriter.WriteStartElement("add_on");
+
```
Comment (joannasese) [inline Library/AddOn.cs]: Given the following update request:
```
addon.UnitAmountInCents["USD"] = 400;
addon.Update(item.ItemCode);
```

request xml is always 

```
<add_on>
  <unit_amount_in_cents>
    <USD>400</USD>
  </unit_amount_in_cents>
  <default_quantity>1</default_quantity>
  <display_quantity_on_hosted_page>false</display_quantity_on_hosted_page>
  <optional>true</optional>
</add_on>
```
when it should be
```
<add_on>
  <unit_amount_in_cents>
    <USD>400</USD>
  </unit_amount_in_cents>
</add_on>
```

File: Library/AddOn.cs [hunk 2]
```diff
@@ -80,6 +88,34 @@ public void Update()
                 ReadXml);
         }
 
+        /// <summary>
+        /// Update an existing item-backed add on in Recurly
+        /// </summary>
+            public void UpdateItemBackedAddOn(
```
Comment (joannasese) [inline Library/AddOn.cs]: A clumsy implementation. Open to other ways to approach this.

Comment (joannasese) [inline Library/AddOn.cs]: A clumsy implementation. Open to other ways to approach this.

File: Library/Plan.cs [hunk 3]
```diff
@@ -171,6 +171,12 @@ public AddOn NewAddOn(string addOnCode, string name)
             return a;
         }
 
+        public AddOn NewAddOn(string itemCode)
+        {
+            var a = new AddOn(PlanCode, itemCode);
+            return a;
```
Comment (bhelx) [inline Library/Plan.cs:177]: no need for a variable here?
```
return new AddOn(PlanCode, itemCode);
```

Comment (joannasese) [reply Library/Plan.cs]: That's a good point.

**Pull request merged**